packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -299,24 +269,65 @@ const EDiscovery = WebexPlugin.extend({
       timeout: options.timeoutMs
     })
       .then((res) => {
-        if (!this.contentSummaryBySpaceIdCache.has(reportId)) {
-          this.contentSummaryBySpaceIdCache.set(reportId, new Map());
-        }
+        const cachePromise = this._writeSpacesToContentSummaryBySpaceIdCache(reportId, [res.body]);
+
+        return Promise.all([cachePromise])
+          .then(() => Promise.resolve(res));
+      });
+  },
+
+  /**
+   * The results of a getContentSummary or getContentSummaryBySpaceId request are written to the contentSummaryBySpaceIdCache
+   * since information for a space is likely to be accessed very frequently when decrypting activities
+   * @param {UUID} reportId - Id of the report which contains the relevant content summary
+   * @param {Array<Object>} spaces - List of the space objects to be written to the cache
+   * @returns {Promise} - Promise resolution indicating operation is complete
+   */
+  _writeSpacesToContentSummaryBySpaceIdCache(reportId, spaces) {
+    if (!reportId || !spaces || !spaces.length) {
+      return Promise.resolve();
+    }
+
+    if (!this.contentSummaryBySpaceIdCache.has(reportId)) {
+      this.contentSummaryBySpaceIdCache.set(reportId, new Map());
+    }
 
-        if (res.body && res.body.spaceId) {
-          try {
-            this.contentSummaryBySpaceIdCache.get(reportId).set(res.body.spaceId, res.body);
-          }
-          catch (error) {
-            this.logger.error(`Error adding ${res.body.spaceId} to contentSummaryBySpaceIdCache: ${error}`);
-          }
+    for (let i = 0; i < spaces.length; i += 1) {
+      const space = spaces[i];
+
+      if (space && space.spaceId) {
+        try {
+          this.contentSummaryBySpaceIdCache.get(reportId).set(space.spaceId, space);
         }
-        else {
-          this.logger.error('Error adding undefined space to contentSummaryBySpaceIdCache');
+        catch (error) {
+          this.logger.error(`Error adding ${space.spaceId} to contentSummaryBySpaceIdCache: ${error}`);
         }
+      }
+      else {
+        this.logger.error('Error adding undefined space to contentSummaryBySpaceIdCache');
+      }
+    }
 
-        return res.body;
-      });
+    return Promise.resolve();
+  },
+
+  @waitForValue('@')
+  /**
+   * Retrieves a config object from the service which can be used by the client for optimal performance, e.g. content page size
+   * @param {Object} options - optional parameters for this method
+   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
+   * @returns {Promise<Config>} Http response containing the config object
+   */
+  getClientConfig(options) {
+    // use spread operator to set default options
+    options = {...defaultOptions, ...options};
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: 'reports/clientconfig',
+      timeout: options.timeoutMs
+    });
```

Comment:
This API will let us dynamically configure aspects of the client behaviour for better performance, for example, we can tweak the page size of the getContentSummary API based on service latency metrics etc.